### PR TITLE
RXR-372: refactor calc_ibw()

### PR DESCRIPTION
**Pull request by karawoo**
`calc_ibw()` had a number of bugs and inconsistencies in its behavior. This PR refactors it into a smaller, less complicated function with a couple of helper functions. 

The main difference in behavior is that `calc_ibw()` no longer provides a vectorized interface. The previous vectorized operations were pretty buggy, and it's really very different data and operations that are getting used depending on the age that's provided to the function. I think this implementation is a lot cleaner and easier to maintain.

I looked at our downstream usage in our internal packages, and from what I can tell it seems we're not using the vectorized interface anyway, so I think this change shouldn't impact those packages. That said, I'm open to trying to bring back support for vector input if we think it's important (though the behavior should probably be different than before; see my comments on the RXR-372 jira ticket).

### CODE REVIEW

File: R/calc_ibw.R [hunk 1]
```diff
@@ -45,75 +48,95 @@ calc_ibw <- function (
...
-      ibw[age >= 1 & age < 18 & ht < 5*12] <- (height[age >= 1 & age < 18 & ht < 5*12]^2 * 1.65)/1000
-      ibw[age >= 1 & age < 18 & ht >= 5*12] <-
-        (39   + 2.27 * (ht[age >= 1 & age < 18 & ht >= 5*12]-(5*12))) * (sex[age >= 1 & age < 18 & ht >= 5*12]=="male") +
-        (42.2 + 2.27 * (ht[age >= 1 & age < 18 & ht >= 5*12]-(5*12))) * (sex[age >= 1 & age < 18 & ht >= 5*12]=="female")
-    }
+  if (age >= 1 & age < 18) {
+    ibw <- switch(
+      method_children,
+      "standard" = ibw_standard(age = age, height = height, sex = sex)
+    )
```
Comment (karawoo) [inline R/calc_ibw.R:76]: The documentation lists some other methods for children that might be implemented in the future, so I put `switch()` here to make it easy to add those, even though for now we only have one method.

File: R/calc_ibw.R [hunk 2]
```diff
@@ -45,75 +48,95 @@ calc_ibw <- function (
...
   ## babies
-  ibw <- rep(0, length(weight))
-  if(any(age < 1)) {
+  if (age < 1) {
     if (is.null(weight)) {
       stop("Actual body weight is used as IBW for children < 1yr. Please supply a weight value.")
     }
     message("Note: using actual body weight as IBW for children < 1yr.")
-    ibw[age < 1] <- weight[age < 1]
+    return(weight)
```
Comment (jasmineirx) [inline R/calc_ibw.R:69]: it's a little odd that ibw is only rounded to `digits` digits if age >= 1. IMO rounding should happen outside of this function and not within it. However, that's better addressed in a separate change. 

made an issue (RXR-405) to track.

**Review by jasmineirx — APPROVED** [review]
this looks great, much more readable! One comment for a future task.

File: R/calc_ibw.R (hunk 2 above)
Comment (karawoo) [reply R/calc_ibw.R:69]: Ah yeah good point. I agree rounding should happen outside the function 👍

**Pull request merged**